client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+        )
+        new_instance = CreatedInstance(
+            self.product_type, product_name, instance_data, self
+        )
+        self._store_new_instance(new_instance)
+
+    def _create_otio_timeline(self, sequence_path, fps):
+        """Creating otio timeline from sequence path
+
+        Args:
+            sequence_path (str): path to sequence file
+            fps (float): frame per second
+
+        Returns:
+            otio.Timeline: otio timeline object
+        """
+        # get editorial sequence file into otio timeline object
+        extension = os.path.splitext(sequence_path)[1]
+
+        kwargs = {}
+        if extension == ".edl":
+            # EDL has no frame rate embedded so needs explicit
+            # frame rate else 24 is assumed.
+            kwargs["rate"] = fps
+            kwargs["ignore_timecode_mismatch"] = True
+
+        return otio.adapters.read_from_file(sequence_path, **kwargs)
+
+    def _get_path_from_file_data(self, file_path_data, multi=False):
+        """Converting creator path data to single path string
+
+        Args:
+            file_path_data (FileDefItem): creator path data inputs
+            multi (bool): switch to multiple files mode
+
+        Raises:
+            FileExistsError: in case nothing had been set
+
+        Returns:
+            str: path string
+        """
+        return_path_list = []
+
+        if isinstance(file_path_data, list):
+            return_path_list = [
+                os.path.join(f["directory"], f["filenames"][0])
+                for f in file_path_data
+            ]
+
+        if not return_path_list:
+            raise FileExistsError(
+                f"File path was not added: {file_path_data}")
+
+        return return_path_list if multi else return_path_list[0]
+
+    def _get_clip_instances(
+        self,
+        folder_entity,
+        otio_timeline,
+        media_folder_path,
+        instance_data,
+        product_type_presets,
+        sequence_file_name,
+        ignore_clip_no_content,
+    ):
+        """Helping function for creating clip instance
+
+        Args:
+            folder_entity (dict[str, Any]): Folder entity.
+            otio_timeline (otio.Timeline): otio timeline object
+            media_folder_path (str): Folder with media files
+            instance_data (dict): clip instance data
+            product_type_presets (list[dict]): list of dict settings
+                product presets
+            sequence_file_name (str): sequence file name
+            ignore_clip_no_content (bool): ignore clips with no content
+        """
+        media_folder_path = media_folder_path.replace("\\", "/")
+
+        # Get all tracks from otio timeline
+        tracks = otio_timeline.video_tracks()
+
+        # get all clipnames from otio timeline to list of strings
+        clip_names = [clip.name for clip in otio_timeline.find_clips()]
+
+        # Create set of clip names for O(1) lookup
+        clip_names_set = set(clip_names)
```

Comment:
`clip_names` is used only for creation of `clip_names_set`, can be skipped.
```suggestion
        clip_names_set = {clip.name for clip in otio_timeline.find_clips()}
```